dali/test/python/sequences_test_utils.py

```diff
@@ -28,13 +29,55 @@
 vid_file = os.path.join(data_root, 'db', 'video',
                         'sintel', 'sintel_trailer-720p.mp4')
 
+
+class SampleDesc:
+
+    def __init__(self, rng: random.Random, frame_idx: int, sample_idx: int, batch_idx: int, sample: np.ndarray):
+        self.rng = rng
+        self.frame_idx = frame_idx
+        self.sample_idx = sample_idx
+        self.batch_idx = batch_idx
+        self.sample = sample
+
+
+class ArgDesc:
+    def __init__(self, name : Union[str, int], is_per_frame : bool, dest_device: str):
+        self.name = name
+        self.is_per_frame = is_per_frame
+        self.dest_device = dest_device
+        assert self.is_positional_arg or dest_device == "cpu", "Named arguments on GPU are not supported"
+
+    @property
+    def is_positional_arg(self):
+        return isinstance(self.name, int)
+
+    def __repr__(self):
+        return "ArgDesc{}".format((self.name, self.is_per_frame, self.dest_device),)
+
+
+class ArgCb:
+    def __init__(self, name: Union[str, int], cb: Callable[[SampleDesc], np.ndarray], is_per_frame: bool, dest_device: str = "cpu"):
+        self.desc = ArgDesc(name, is_per_frame, dest_device)
+        self.cb = cb
+
+    def __repr__(self):
+        return "ArgCb{}".format((self.cb, self.desc))
+
+
+class ArgData:
+    def __init__(self, arg_desc : ArgDesc, data):
+        self.desc = arg_desc
+        self.data = data
+
+
 class ParamsProviderBase:
     """
     Computes data to be passed as argument inputs in sequence processing tests, the `compute_params` params
-    should return tuple of lists with data for respectively per-sample and per-frame arguments.
-    The `expand_params` should return corressponding unfolded/expanded arguments to be used in the
+    should return a lists of ArgData; the `arg_desc.data` dimensionality must match `arg_data.desc.is_per_frame`.
```

Comment:
```suggestion
    should return a lists of ArgData; number of dimensions in the `arg_desc.data` and `arg_data.desc.is_per_frame` arguments must match.
```

Reply:
changed it to
"should return a lists of ArgData; number of dimensions in the `arg_desc.data` must reflect `arg_data.desc.is_per_frame` argument."

Reply:
Because, in the hindsight maybe match is not the most fortunate word in this context.